pkg/watchers/apiserver/apiserver.go

```diff
@@ -16,34 +16,38 @@ import (
 	fm "github.com/microsoft/retina/pkg/managers/filtermanager"
 	"github.com/microsoft/retina/pkg/pubsub"
 	"go.uber.org/zap"
+	"k8s.io/client-go/rest"
 	kcfg "sigs.k8s.io/controller-runtime/pkg/client/config"
 )
 
 const (
 	filterManagerRetries = 3
+	hostLookupRetries    = 3
 )
 
 type ApiServerWatcher struct {
-	isRunning     bool
-	l             *log.ZapLogger
-	current       cache
-	new           cache
-	apiServerUrl  string
-	hostResolver  IHostResolver
-	filtermanager fm.IFilterManager
+	isRunning         bool
+	l                 *log.ZapLogger
+	current           cache
+	new               cache
+	apiServerHostName string
+	hostResolver      IHostResolver
+	filterManager     fm.IFilterManager
+	restConfig        *rest.Config
+	remainingRetries  int
```

Comment:
Discussed offline; this seems confusing. Refactor please.
- either use a timestamp to save last success, or
- Use the [retry](https://github.com/microsoft/retina/blob/c7e4e6455eb5d6cc876dd5ec13b50019731a7271/pkg/utils/common.go#L46) util function